src/expand.rs

```diff
@@ -64,6 +92,12 @@ pub fn expand(input: &mut Item, is_local: bool) {
                     if sig.asyncness.is_some() {
                         let block = &mut method.default;
                         let mut has_self = has_self_in_sig(sig);
+                        let method_args = validate_method_attrs(
```

Comment:
I was on the fence about doing the attr matching in `parse` or here. If I move it to `parse.rs` the changes will be considerably larger I think, but there'd be a clearer separation between parsing and expanding and I could revert the return type for `expand` from `Result` to `()`